cocotb/regression.py

```diff
@@ -308,8 +310,17 @@ def _init_test(self, test: Test) -> Optional[RunningTask]:
             self._record_result(test, test_init_outcome, 0, 0)
             return None
 
-        test = test_init_outcome.get()
-        return test
+        running_test = test_init_outcome.get()
+
+        # seed random number generator based on test module, name, and RANDOM_SEED
+        hasher = hashlib.sha1()
+        hasher.update(test.__qualname__.encode())
+        hasher.update(test.__module__.encode())
```

Comment:
As I commented, it's not unusual where I work for people to collect failing tests from many different test files (MODULE files) into one big regression MODULE file.  Users will often call this file "The Kitchen Sink" or something creative like that. :-)  If the MODULE name is used here then users would not be able to do that  i.e. if they had a hundred failing tests they could potentially be forced to have a hundred separate MODULE files as well... rather than one.  Not a deal breaker but I thought I should mention it.

Reply:
`__module__` will be the name of the module they were defined in, not the module they were discovered in. So importing tests from other modules into a "kitchen sink" module will continue to work.